addons/account/models/account_payment.py

```diff
@@ -169,33 +228,19 @@ def _compute_payment_difference(self):
 
     @api.onchange('journal_id')
     def _onchange_journal(self):
-        if self.journal_id:
-            if self.journal_id.currency_id:
-                self.currency_id = self.journal_id.currency_id
-
-            # Set default payment method (we consider the first to be the default one)
-            payment_methods = self.payment_type == 'inbound' and self.journal_id.inbound_payment_method_ids or self.journal_id.outbound_payment_method_ids
-            payment_methods_list = payment_methods.ids
-
-            default_payment_method_id = self.env.context.get('default_payment_method_id')
-            if default_payment_method_id:
-                # Ensure the domain will accept the provided default value
-                payment_methods_list.append(default_payment_method_id)
-            else:
-                self.payment_method_id = payment_methods and payment_methods[0] or False
-
-            # Set payment method domain (restrict to methods enabled for the journal and to selected payment type)
-            payment_type = self.payment_type in ('outbound', 'transfer') and 'outbound' or 'inbound'
+        if not self.journal_id:
+            return
 
-            domain = {'payment_method_id': [('payment_type', '=', payment_type), ('id', 'in', payment_methods_list)]}
+        if self.journal_id.currency_id:
+            self.currency_id = self.journal_id.currency_id
 
-            if self.env.context.get('active_model') == 'account.move':
-                active_ids = self._context.get('active_ids')
-                invoices = self.env['account.move'].browse(active_ids)
-                self.amount = abs(self._compute_payment_amount(invoices, self.currency_id, self.journal_id, self.payment_date))
+        if not self.env.context.get('default_payment_method_id'):
+            self.payment_method_id = self._payment_methods[:1]._origin
```

Comment:
Maybe put deault_payment_method as the first of _payment_methods and always do this?

Reply:
not sure that's correct: we probably don't want to reset the payment method to the default every time the journal_id changes, or do we?

Reply:
(though why does this run on journal_id being changed but not any other value used in this method e.g. payment type or currency?)